src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableCommonContext.cs

```diff
@@ -372,32 +368,64 @@ internal virtual void ThrowIfInvalidAccess()
-                    this.Config.TraceHelper.FunctionFailed(
-                        this.Config.Options.HubName,
-                        functionName,
-                        this.InstanceId,
-                        operationId,
-                        operationName,
-                        reason: $"(replayed {exception.GetType().Name})",
-                        functionType: functionType,
-                        isReplay: true);
+
+                    if (functionType == FunctionType.Entity)
+                    {
+                        this.Config.TraceHelper.OperationCompleted(
+                            this.Config.Options.HubName,
+                            functionName,
+                            this.InstanceId,
+                            operationId,
+                            operationName,
+                            input: "(replayed)",
+                            output: "(replayed)",
+                            failed: true,
+                            duration: 0,
+                            isReplay: true);
+                    }
+                    else
+                    {
+                        this.Config.TraceHelper.FunctionFailed(
+                            this.Config.Options.HubName,
+                            functionName,
+                            this.InstanceId,
+                            reason: $"(replayed {exception.GetType().Name})",
+                            functionType: functionType,
+                            isReplay: true);
+                    }
                 }
             }
 
             if (this.IsReplaying)
             {
                 // If this were not a replay, then the orchestrator/activity/entity function trigger would have already
                 // emitted a FunctionCompleted trace with the actual output details.
-                this.Config.TraceHelper.FunctionCompleted(
+
+                if (functionType == FunctionType.Entity)
+                {
+                    this.Config.TraceHelper.OperationCompleted(
+                        this.Config.Options.HubName,
+                        functionName,
+                        this.InstanceId,
+                        operationId,
+                        operationName,
+                        input: "(replayed)",
+                        output: "(replayed)",
+                        failed: false,
+                        duration: 0,
+                        isReplay: true);
+                }
+                else
+                {
+                    this.Config.TraceHelper.FunctionCompleted(
                     this.Config.Options.HubName,
                     functionName,
                     this.InstanceId,
-                    operationId,
-                    operationName,
                     output: "(replayed)",
                     continuedAsNew: false,
                     functionType: functionType,
                     isReplay: true);
```

Comment:
nit: need one more level of indentation for the parameter list.

Reply:
yes.